### PR TITLE
Upgrade Phoenix to 1.4.1

### PR DESCRIPTION
**Pull request by codyduval**
Upgrade Phoenix and its related dependencies to version 1.4.1,
primarily to address race conditions within Phoenix Presence.

__Description of Change(s) Introduced:__ 
- Upgrade to Cowboy 2, which includes support for HTTP2
- Replace Poison with Jason for json generation
- Remove calls to transport macro, new api for socket config
- Refactor presence on client side to comply with new Presence API
- Update formatter config to format routing and test DSLs
- Alias Router.Helpers (instead of import) to remove compile-time dep
- Update circleci cache keys to force builds with new deps
- Fix bug in user_by_id reducer that was causing errors when
  users joined a retro.

__Dependencies Introduced or Upgraded:__ 

Backend:
- `phoenix` 1.3.4 ---> 1.4.1
- `ecto_sql` 3.0 ---> 3.0.5
- `cowboy` 1.0 ---> `plug_cowboy` 2.0.1
- `plug` 1.7.2
- `jason` 1.1.2
- `poison` removed

Client:
- New `phoenix.js` included in phoenix package

__Relevant github Issue:__ 

- [Issue #519](https://github.com/stride-nyc/remote_retro/issues/519)

### CODE REVIEW

**Comment by codyduval** [comment]
@vanderhoop I'll note that this should be heavily click-tested in some sort of staging environment before merging.

File: .env-sample [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export REMOTE_RETRO_GOOGLE_OAUTH_CLIENT_ID="<Client Id>"
```
Comment (vanderhoop) [inline .env-sample]: can we kill this? or include it in a separate PR with an explanation of its use? I'm not 100% against it; it just feels unrelated

File: .formatter.exs [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
-  inputs: [
-    "mix.exs",
-    "{config,lib,test}/**/*.{ex,exs}"
-  ],
+  import_deps: [:phoenix],
+  inputs: ["mix.exs", "*.{ex,exs}", "{config,lib,priv,test}/**/*.{ex,exs}"],
   line_length: 120,
   locals_without_parens: [],
-  trailing_comma: true,
+  trailing_comma: true
```
Comment (vanderhoop) [inline .formatter.exs]: the irony here is too good! love it!

File: .gitignore [hunk 3]
```diff
@@ -47,3 +47,5 @@ package-lock.json
 npm-debug.log*
 .eslintcache
 browser_logs.log
+# Local secret env vars
```
Comment (vanderhoop) [inline .gitignore]: see comment above. also, if you want to make a case for inclusion in _this_ PR, i'm down to hear it!

File: .env-sample (hunk 1 above)
Comment (vanderhoop) [reply .env-sample]: r, if you want to make a case for inclusion in this PR, i'm down to hear it!

File: web/static/js/services/retro_channel.js [hunk 4]
```diff
@@ -11,27 +11,42 @@ class RetroChannel {
   }
 }
 
+export const handlePresence = (presence, setPresences) => {
```
Comment (vanderhoop) [inline web/static/js/services/retro_channel.js]: are we introducing this helper due to ease of testing? it almost feels like the indirection of passing the `presence` and `setPresences` action makes this a little less grokkable? 

what if we kept it inline down below?
```es6
const presence = new Presence(retroChannel)

presence.onSync(() => {
    const users = presence.list((_token, presence) => (presence.user))
    setPresences(users)
  })


```

File: test/redux/ideas_test.js [hunk 5]
```diff
@@ -146,11 +146,12 @@ describe("actionCreators", () => {
 
         it("dispatches an error", () => {
           push = mockRetroChannel.push("anyEventJustNeedThePushInstance", { foo: "bar" })
+          push.trigger("ok", {})
```
Comment (vanderhoop) [inline test/redux/ideas_test.js]: this is the only thing that gets me, as i can't understand *why* this is suddenly necessary. gonna peep through the phoenix source to see if there's something we can isolate in the `setupMockPhoenixChannel` helper

File: .env-sample (hunk 1 above)
Comment (codyduval) [reply .env-sample]: Oh not a biggie, it can go elsewhere.

File: .gitignore (hunk 3 above)
Comment (codyduval) [reply .gitignore]: 👍

File: web/static/js/services/retro_channel.js (hunk 4 above)
Comment (codyduval) [reply web/static/js/services/retro_channel.js]: Yeah - originally it was more complicated, but now that it's a two liner, makes sense to get rid of it (I also thought I would test it separately, but that was also unneeded).

File: test/redux/ideas_test.js (hunk 5 above)
Comment (codyduval) [reply test/redux/ideas_test.js]: I know - I decided to ship it after too much time in the rabbit hole.  Would be great if you could provide insight.

Comment (vanderhoop) [reply test/redux/ideas_test.js]: yeah, i've been rabbit holing for awhile, and it's baby steps

Comment (vanderhoop) [reply test/redux/ideas_test.js]: by which i mean, slow and unsteady progress

**Comment by vanderhoop** [comment]
great work on this, @codyduval! I'm going to merge this over the weekend, as it's a significant upgrade and changeset.
